### PR TITLE
rmount: init at 1.0.1

### PR DESCRIPTION
**Pull request by Qubasa**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
-  [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/filesystems/rmount/default.nix
Comment (samueldr) [inline pkgs/tools/filesystems/rmount/default.nix]: [From `makeWrapper`'s documentation](https://nixos.org/nixpkgs/manual/#ssec-stdenv-functions):

> `wrapProgram` is a convenience function you probably want to use most of the time.

In this particular instance, unless it is expected for a end-user to use `rmount-noenv`, you probably would prefer to use `wrapProgram`.

> `wrapProgram` *`executable`* *`makeWrapperArgs`*
> Convenience function for makeWrapper that automatically creates a sane wrapper file It takes all the same arguments as `makeWrapper`, except for `--argv0`.

File: pkgs/tools/filesystems/rmount/default.nix
Comment (samueldr) [inline pkgs/tools/filesystems/rmount/default.nix]: Derivations in `nixpkgs` *never* use `pkgs`, and neither would `import <nixpkgs>`. `mkDerivation` and `callPackage` handles dependency injection.

Additionally, using `with` is an issue, as here it makes everything available without proper dependency injection.

File: pkgs/tools/filesystems/rmount/default.nix
Comment (samueldr) [inline pkgs/tools/filesystems/rmount/default.nix]: [You probably want to add yourself as maintainer (and to maintainers.nix)](https://nixos.org/nixpkgs/manual/#chap-meta). (Doubly so since in addition to maintaining it here, you seem to be the author!)

You also will need to provide the `platforms` attribute. 

> Hydra builds packages according to the platform specified. If no platform is specified, the package does not have prebuilt binaries.

File: pkgs/tools/filesystems/rmount/default.nix
Comment (samueldr) [inline pkgs/tools/filesystems/rmount/default.nix]: `stdenv` doesn't need to be given to `buildInputs`.

**Review by samueldr — CHANGES_REQUESTED** [review]
:wave: Hi! Thanks for your first contribution!

Just a couple things to fix, nothing serious. If you have any questions, hesitations, don't hesitate and ask.

**Comment by samueldr** [comment]
Additionally, it may or may not be an issue, the git commit's author isn't properly linked to the github account; it could either be that you didn't add that e-mail address in your github account (not really an issue) or misconfigured git (maybe more of an issue). (The e-mail address used seems a bit... placeholder~ish.)

**Comment by Luis-Hebendanz** [comment]
Thank you very much for your help. I fixed every issue you described, the email was intentionally  placeholder~ish but to eliminate conflicts, I just changed it to a real one.

**Comment by Luis-Hebendanz** [comment]
Should I close this Pull request and reopen a new one?

File: pkgs/tools/filesystems/rmount/default.nix
Comment (samueldr) [inline pkgs/tools/filesystems/rmount/default.nix]: With GitHub, it is preferred to use fetchFromGitHub, since it will make use of GitHub specific end-points to fetch the revision in a way that's (1) safe from their sometimes-changing archives (2) better-suited to fetch than plain git. Its use is pretty much like fetchgit, except that you pass `owner` and `repo` instead of `url`.

File: pkgs/tools/filesystems/rmount/default.nix
Comment (samueldr) [inline pkgs/tools/filesystems/rmount/default.nix]: `lib` can be removed from the parameters up top, and `stdenv.lib.makeBinPath` used here.

In Nixpkgs there are 205 `stdenv.lib.makeBinPath` and 109 `lib.makeBinPath`, making the former preferred.

If you're curious, `stdenv.lib` is the same as `lib`.

```
nix-repl> stdenv.lib == lib
[...]
true
```

**Review by samueldr — COMMENTED** [review]
No need to close it, you could force push on your branch, with one squashed commit, that commit having the right author information :).

Additionally, new notes, one I missed initially.

**Comment by Luis-Hebendanz** [comment]
Thanks! Now it's squashed and the minor things are fixed :-)

File: pkgs/tools/filesystems/rmount/default.nix
Comment (aanderse) [inline pkgs/tools/filesystems/rmount/default.nix]: Please move this declaration directly into the `fetchFromGitHub` call instead of calling `inherit rev;`

File: pkgs/tools/filesystems/rmount/default.nix
Comment (aanderse) [inline pkgs/tools/filesystems/rmount/default.nix]: Please change to `pname = "rmount";`

**Review by aanderse — CHANGES_REQUESTED** [review]

**Comment by aanderse** [comment]
@Luis-Hebendanz in addition to the 2 requests I have made could you please rewrite the git history to contain 2 commits: the first simply adding yourself to the list of maintainers with an appropriate commit message, and the second commit adding the package?

After these changes we'll get this merged right away.

**Comment by Luis-Hebendanz** [comment]
@aanderse Closing this branch I opened up a new one with desired changes: https://github.com/NixOS/nixpkgs/pull/60204

**Pull request closed without merging**
